plugins/repository-azure/src/main/java/org/opensearch/repositories/azure/AzureStorageSettings.java

```diff
@@ -221,6 +232,7 @@ private AzureStorageSettings(
     ) {
         this.account = account;
         this.connectString = connectString;
```

Comment:
We need to remove `connectString` and use the strategy based on the `tokenCredentialType`, for example:

```
    private final Function<BlobServiceClientBuilder, BlobServiceClientBuilder> clientBuilder;


    private AzureStorageSettings( .... ) {
        ...
        if (this.tokenCredentialType != null) {
            this.clientBuilder = (builder) -> builder.connectionString(buildConnectString(account, key, sasToken, endpointSuffix));
        } else {
            this.clientBuilder = (builder) -> builder.credential(new ManagedIdentityCredentialBuilder()).endpoint(getStorageBlobEndpoint().getPrimaryUri());
        }
}

public BlobServiceClientBuilder configure(BlobServiceClientBuilder builder) {
    return clientBuilder.apply(builder); 
}
```

Reply:
(the usage of configure method is here https://github.com/opensearch-project/OpenSearch/pull/12559/files#r1570795810

Reply:
refactored